### DISCUSSION TITLE
ENFUGUE Server v0.1.2

### DISCUSSION THREAD

**Discussion by painebenjamin** [painebenjamin/app.enfugue.ai#26] [Announcements]
# Thank you!

Thank you to everyone who has helped test so far, you've all been extremely helpful. 

I hope this release corrects a lot of the issues people have been having!

# Installation
### First-Time Installation
```sh
pip install enfugue
```

If you are on Linux and want TensorRT support, execute:
```
pip install enfugue[tensorrt]
```

If you are on Windows and want TensorRT support, follow the steps detailed [here.](https://github.com/painebenjamin/app.enfugue.ai#windows-tensorrt-support)

### Upgrading from 0.1.x
```sh
pip install enfugue --upgrade
```

### Linux
Download the `manylinux` files here, concatenate them and extract them. A simple console command to do that is:

```sh
cat enfugue-server-0.1.2*.part | tar -xvz
```
### Windows
Download the `win64` files here, and extract them using a program which allows extracting from multiple archives such as [7-Zip](https://www.7-zip.org/). 

**If you are using 7-Zip, you should _not_  extract both files independently. If they are in the same directory when you unzip the first, 7-Zip will automatically unzip the second. The second file cannot be extracted on it's own.**

# New Features

- Added directory options to initialization, allowing you to set where Enfugue looks for and stores checkpoints, LoRA, and other models.
  - Note: Enfugue will only create a directory if it is in it's base cache directory. If you specify a directory outside of that, and the directory does not exist, Enfugue will not accept the input.
- Added a "change directory" option in `System > Installation Manager` to change directories after initialization.
  - Note: Files are not moved when you do this. If you want to bring the files from the old directory to the new, you will need to copy them over yourself.
- Added a new menu option `System > Engine Logs`. This gives you a realtime view of the activities of the diffusion engine, which inclues all activities of Stable Diffusion itself, as well as any necessary downloads or longer-running processes like TensorRT engine builds.
  - Note: This is a real-time view, and will always show the most recent 100 log entries. There can be a lot of logs, so it's necessary to trim the UI often or else it will bog down substantially. If you want to view the logs in non-real-time, navigate to the your .cache directory (in `/home/<youruser>` on linux, `C:\Users<youruser>.cache` on windows, substitute your drive letter as needed.)
- Added a new command dump-config that reads the packaged configuration a writes to stdout or a file. Default format is yaml, but json is also supported.
```
Usage: enfugue dump-config [OPTIONS]

  Dumps a copy of the configuration to the console or the specified path.

Options:
  -f, --filename TEXT  A file to write to instead of stdout.
  -j, --json           When passed, use JSON instead of YAML.
  --help               Show this message and exit.
```
- Added a new flag to run that allows you to specify a configuration file to load instead of the default. enfugue run now has the signature:
```
Usage: enfugue run [OPTIONS]

  Runs the server synchronously using cherrypy.

Options:
  -c, --config TEXT  An optional path to a configuration file to use instead
                     of the default.
  --help             Show this message and exit.
```
  - Note: The configuration file must have a proper extension indicating it's format, i.e. either /json or .yml/.yaml.

[Documentation regarding what settings are available and what they do is up on the wiki.](https://github.com/painebenjamin/app.enfugue.ai/wiki/Configuration-for-Advanced-Users)

# Issue Fixes

- Fixed an issue where JavaScript Module files were being served with the inappropriate Content-Type, resulting in a non-functional UI.
- Fixed an issue where the base Stable Diffusion model would be initialized twice when it was explicitly being used, consuming singificant extra amounts of VRAM.
- Fixed an issue where the Polygraphy package was listed as always required, when it is only required for TensorRT

# Changes

- Removed need to create diffusers cache directory when not using TensorRT (saves significant hard disk space)
- Added output to the enfugue run command so you know it's working and what URL to go to.
- Separated server logs and engine logs. Server logs are now kept at the previous `~/.cache/enfugue.log` location, and engine logs are at `~/.cache/enfugue-engine.log`
  - Server logs have had their default level changed to `ERROR` to hide unhelpful messages, as the server is mostly stable.
  - Engine logs have their default level at `DEBUG` to give as much information as possible to the front-end. This may change in the future.

<hr /><em>This discussion was created from the release <a href='https://github.com/painebenjamin/app.enfugue.ai/releases/tag/0.1.2'>ENFUGUE Server v0.1.2</a>.</em>

**Comment by thesomeotherguy** [comment]
can standalone Windows user get the update feature too?

downloading 3GBs every update seems not efficient way to handle an update.

or there's a way I didn't know how to?

**Comment by painebenjamin** [comment]
Hello!

I agree wholeheartedly, and the next update will have a much smaller
package for standalone users. Unfortunately for this one standalone users
will have to re-download.

Sorry about that. Python's package manager does a lot of the heavy lifting
of managing what needs updating and what doesn't, and when it's standalone
we lose that benefit. That just means we need to do that ourselves, so
it'll take a bit to orchestrate that, but it will be done for the next one.

I'm also hoping to get a little more time in between updates, but the fixes
in this one were critical for a lot of people.

Thank you for your message!

**Reply by thesomeotherguy** [reply]
okay.. if that's the case.

maybe I should learn the pip installation next time, it's using venv then ./activate right?

**Comment by painebenjamin** [comment]
I highly recommend using Conda, as this will also manage binary
dependencies. If you don't use Conda, you'll have to manually install CUDA
at a minimum, and make sure it's available on your system path - it's way
easier to just let an environment manager do it for you.

The instructions for that are at:
https://github.com/painebenjamin/app.enfugue.ai#advanced-creating-your-own-environment-and-running-from-command-line

**Reply by thesomeotherguy** [reply]
thanks.

will learn installing it from conda. I search information about what conda is, and it's like venv but more advanced and easy to manage.

for cuda, I already install it separately with installer from Nvidia and also cuDNN extracted into CUDA installation directory.
